apps/src/lib/node/ledger/shell/prepare_proposal.rs

```diff
@@ -238,30 +238,55 @@ mod prepare_block {
             Some(VoteExtensionDigest { events, signatures })
         }
 
+        /// Takes a list of signed vote extensions,
+        /// and filters out invalid instances, returning
+        /// all residual values (including invalid vote
+        /// extensions).
+        #[inline]
+        pub fn filter_invalid_vote_extensions_residuals(
```

Comment:
Can this function go live in the `vote_extensions` module? It makes sense there and will tighten up this file a bit.